apps/web/app/(app)/[emailAccountId]/assistant/RuleForm.tsx

```diff
@@ -1226,6 +1240,68 @@ function ActionCard({
           {/* Show the variable pro tip when action has visible fields that can use variables */}
           {shouldShowProTip && <VariableProTip />}
 
+          {/* Show delay toggle and input for all actions that support delays */}
+          {canActionBeDelayed(action.type) && (
+            <div className="mt-4">
+              <div className="flex items-start justify-between">
+                <div className="flex-1">
+                  {/* Show delay input controls when delay has been enabled */}
+                  {watch(`actions.${index}.delayInMinutes`) !== null && (
+                    <DelayInputControls
+                      index={index}
+                      delayInMinutes={watch(`actions.${index}.delayInMinutes`)}
+                      setValue={setValue}
+                    />
+                  )}
+                </div>
+
+                <div className="flex items-center space-x-2">
+                  <TooltipExplanation
+                    text="Schedule this action to execute after a delay from when the email was received. Useful for follow-ups, reminders, or giving senders time to respond."
+                    side="right"
+                  />
+                  <Toggle
+                    name={`actions.${index}.delayEnabled`}
+                    label="Delay"
+                    enabled={watch(`actions.${index}.delayInMinutes`) !== null}
+                    onChange={(enabled: boolean) => {
+                      if (enabled) {
+                        setValue(`actions.${index}.delayInMinutes`, 60, {
+                          shouldValidate: true,
+                        });
+                      } else {
+                        setValue(`actions.${index}.delayInMinutes`, null, {
+                          shouldValidate: true,
+                        });
+                      }
+                    }}
+                  />
+                </div>
+              </div>
+
+              {errors?.actions?.[index]?.delayInMinutes && (
+                <div className="mt-2">
+                  <ErrorMessage
+                    message={
+                      errors.actions?.[
+                        index
+                      ]?.delayInMinutes?.message?.toString() ||
+                      "Please enter a valid delay between 1 minute and 90 days"
```

Comment:
zod can handle this error message for us. It will automatically, but if you want to customise the text you can do so too. No need to do it here.

More: https://zod.dev/error-formatting